src/views/dialogs/DepositDialog2/utils.ts

```diff
@@ -169,7 +169,10 @@ export function useDepositSteps({
                   abi: ERC20ABI,
                   functionName: 'approve',
                   args: [approvalMaybeNeeded.spender as Address, maxUint256],
-                  chain: signer.chain,
+                  chain:
+                    CHAIN_ID_TO_INFO[
+                      Number(approvalMaybeNeeded.chainId) as keyof typeof CHAIN_ID_TO_INFO
```

Comment:
create re-usable typecasted chainId to index VIEM_PUBLIC_CLIENTS and CHAIN_ID_TO_INFO?

Reply:
both keys are the sameeeeee

Reply:
ohh so true bestiee